ledger/participant-integration-api/src/main/scala/platform/apiserver/LedgerApiServer.scala

```diff
@@ -65,6 +65,11 @@ private[daml] final class LedgerApiServer(
         override def servicesClosed(): Future[Unit] =
           servicesClosedPromise.future
       }
+    }).transformWith {
+      case Failure(ex) =>
```

Comment:
Why do we need to log explicitly here? Isn't the Resource going to bubble up the failure all the way through the chain?

Reply:
So here is how they bubble up. Unless the exception is a `SuppressedStartupException`, I don't see how it would skip logging.

Reply:
I think it skips because of timing, the log is protracted until the very end of the shutdown and is not printed when something else pops up in between

Reply:
This is not purely theoretical, I have been able to reproduce it many times

Reply:
>I think it skips because of timing, the log is protracted until the very end of the shutdown and is not printed when something else pops up in between

I see. The possibility of skipping failure logs on application startup is not ideal so we should revisit this part at some point since we perform a lot of startup operation in these resources. Nice findings. LGTM 👍